jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/URIUtil.java

```diff
@@ -1372,8 +1385,12 @@ public static boolean isValidHostRegisteredName(String token)
     public static String newURI(String scheme, String server, int port, String path, String query)
     {
         StringBuilder builder = newURIBuilder(scheme, server, port);
-        builder.append(path);
-        if (query != null && query.length() > 0)
+        boolean hasQuery = !StringUtil.isBlank(query);
+        if (!StringUtil.isBlank(path))
+            builder.append(path);
+        else if (hasQuery)
+            builder.append('/');
+        if (hasQuery)
             builder.append('?').append(query);
```

Comment:
This is a different output than HttpURI now.

The input `("http", "local", 8080, null, "a=b")`

Results in `http://local:8080?a=b` in HttpURI.asString().
But will result in `http://local:8080/?a=b` in URIUtil with this change.

Reply:
Looking at other implementations, they all accept no declared, but treat them differently.
I'm inclined to think the HttpURI versions is wrong, and we should emit a path.

**java.net.URL**

```
jshell> var u = new URL("http://local:8080?a=b")
u ==> http://local:8080?a=b

jshell> u.getPath()
$2 ==> ""

jshell> u.getQuery()
$3 ==> "a=b"
```

**java.net.URI**

```
jshell> var u = new URI("http://local:8080?a=b")
u ==> http://local:8080?a=b

jshell> u.getPath()
$2 ==> ""

jshell> u.getQuery()
$3 ==> "a=b"
```

**Chrome**

```
var u = new URL("http://local:8080?a=b")
  u URL {origin: 'http://local:8080', protocol: 'http:', username: '', password: '', host: 'local:8080', …}
    hash: ""
    host: "local:8080"
    hostname: "local"
    href: "http://local:8080/?a=b"
    origin: "http://local:8080"
    password: ""
    pathname: "/"
    port: "8080"
    protocol: "http:"
    search: "?a=b"
    searchParams: URLSearchParams {size: 1}
    username: ""
    [[Prototype]]: URL
```
    
**Firefox**

```
var u = new URL("http://local:8080?a=b")

u URL { 
  href: "http://local:8080/?a=b", 
  origin: "http://local:8080", 
  protocol: "http:", 
  username: "", 
  password: "", 
  host: "local:8080", 
  hostname: "local", 
  port: "8080", 
  pathname: "/", 
  search: "?a=b" 
}
```

Reply:
Tests and fixes incoming ...